internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -406,8 +406,8 @@ func suggestChangeWaitTimeout(err error, flagName string) error {
 
 func (md *machineDeployment) waitForMachine(ctx context.Context, e *machineUpdateEntry) error {
 	lm := e.leasableMachine
-	// Don't wait for SkipLaunch machines, they are updated but not started
-	if e.launchInput.SkipLaunch {
+	// Don't wait for SkipLaunch or scheduled machines, they are updated but not started
+	if e.launchInput.SkipLaunch || e.launchInput.Config.Schedule != "" {
```

Comment:
setting SkipLaunch=true is the right way to do it unless there is something else special about Schedule, see https://github.com/superfly/flyctl/blob/master/internal/command/deploy/machines_launchinput.go#L205